packages/main/src/TableVirtualizer.ts

```diff
@@ -0,0 +1,279 @@
+
+/**
+ * Fired when the virtualizer is changed by user interaction e.g. on scrolling.
+ *
+ * @public
+ */
+@event<RangeChangeEventDetail>("range-change", {
+	detail: {
+		/**
+		 * The 0-based index of the first children currently rendered.
+		 * @public
+		 */
+		first: { type: Number },
+		/**
+		 * The 0-based index of the last children currently rendered.
+		 * @public
+		 */
+		last: { type: Number },
+	},
+})
+
+class TableVirtualizer extends UI5Element implements ITableFeature {
+	/**
+	 * Defines the height of the rows in the table.
+	 *
+	 * <b>Note:</b> This property is mandatory for the virtualization to work properly.
+	 *
+	 * @default 45
+	 * @public
+	 */
+	@property({ type: Number })
+	rowHeight = 45;
+
+	/**
+	 * Defines the total count of rows in the table.
+	 *
+	 * <b>Note:</b> This property is mandatory for the virtualization to work properly.
+	 *
+	 * @default 100
+	 * @public
+	 */
+	@property({ type: Number })
+	rowCount = 100;
+
+	/**
+	 * Defines the count of extra rows to be rendered in the table.
+	 *
+	 * @default 0
+	 * @public
+	 */
+	@property({ type: Number })
+	extraRows = 0;
+
+	readonly identifier = "TableVirtualizer";
+
+	_table?: Table;
+	_lastRowPosition: number = 0;
+	_firstRowPosition: number = 0;
+	_visibleRowCount: number = 0;
+	_tabBlockingState: TabBlocking = TabBlocking.None;
+	_onRowInvalidateBound: (invalidationInfo: InvalidationInfo) => void;
+	_onScrollBound: () => void;
+
+	constructor() {
+		super();
+		this._onScrollBound = throttle(this._onScroll.bind(this));
+		this._onRowInvalidateBound = this._onRowInvalidate.bind(this);
+	}
+
+	onTableActivate(table: Table): void {
+		this._table = table;
+		this._updateRowsHeight();
+		this._scrollContainer.addEventListener("scroll", this._onScrollBound, { passive: true });
+		this._onScroll();
+	}
+
+	onAfterRendering(): void {
+		this._table && this._table._invalidate++;
+	}
+
+	onTableAfterRendering(): void {
+		if (!this._table) {
+			return;
+		}
+
+		this._updateRowsHeight();
+		if (this._tabBlockingState & TabBlocking.Released) {
+			const tabBlockingRow = this._table.rows.at(this._tabBlockingState & TabBlocking.Next ? -1 : 0) as HTMLElement;
+			const tabForwardingElement = getTabbableElements(tabBlockingRow).at(this._tabBlockingState & TabBlocking.Next ? 0 : -1);
+			this._tabBlockingState = TabBlocking.None;
+			(tabForwardingElement || tabBlockingRow).focus();
+		}
+	}
+
+	onExitDOM(): void {
+		this._scrollContainer.removeEventListener("scroll", this._onScrollBound);
+		this._table = undefined;
+	}
+
+	/**
+	 * Resets the virtualizer to its initial state and triggers the <code>range-change</code> event.
+	 * @public
+	 */
+	reset(): void {
+		this._lastRowPosition = -1;
+		this._firstRowPosition = -1;
+		if (this._scrollContainer.scrollTop > 0) {
+			this._scrollContainer.scrollTop = 0;
+		} else {
+			this._onScroll();
+		}
+	}
+
+	get _scrollContainer() {
+		return this._table!._tableElement;
```

Comment:
Is this working for something like this?

```html
<div style="height: 375px; overflow: auto">
  <ui5-table>
    <ui5-table-virtualizer row-height="51" row-count="1000" slot="features"></ui5-table-virtualizer>
  </ui5-table>
</div>
```

If that is not the case, we might need to explicitly document the restrictions

Reply:
Forget what I wrote, it does not apply anymore